Tests/ApolloCodegenTests/CodeGeneration/Templates/SelectionSet/SelectionSetTemplateTests.swift

```diff
@@ -1969,6 +1969,806 @@ class SelectionSetTemplateTests: XCTestCase {
+    """
+
+    document = """
+    query TestOperation($filter: Boolean) {
+      allAnimals {
+        ... on Dog @skip(if: $filter) @defer {
+          species
+        }
+      }
+    }
+    """
+
+    let expected = """
+      public static var __selections: [ApolloAPI.Selection] { [
+        .field("__typename", String.self),
+        .include(if: !"filter", .inlineFragment(AsDogIfNotFilter.self, deferred: true)),
+      ] }
+    """
+
+    // when
+    try buildSubjectAndOperation()
+
+    let allAnimals = try XCTUnwrap(operation[field: "query"]?[field: "allAnimals"] as? IR.EntityField)
+    let actual = subject.render(field: allAnimals)
+
+    // then
+    expect(actual).to(equalLineByLine(expected, atLine: 7, ignoringExtraLines: true))
+  }
+
+  func test__render_selection__givenInlineFragment_withDeferDirective_andIfArgument_andConditionalDirective_rendersConditionalSelection_withDeferredInlineFragmentSelectionWithVariable() throws {
+    // given
+    schemaSDL = """
+    type Query {
+      allAnimals: [Animal!]
+    }
+
+    interface Animal {
+      species: String!
+    }
+
+    type Dog implements Animal {
+      species: String!
+    }
+    """
+
+    document = """
+    query TestOperation($filter: Boolean, $other: Boolean) {
+      allAnimals {
+        ... on Dog @include(if: $filter) @defer(if: $other) {
+          species
+        }
+      }
+    }
+    """
+
+    let expected = """
+      public static var __selections: [ApolloAPI.Selection] { [
+        .field("__typename", String.self),
+        .include(if: "filter", .inlineFragment(AsDogIfFilter.self, deferred: .if("other"))),
+      ] }
+    """
+
+    // when
+    try buildSubjectAndOperation()
+
+    let allAnimals = try XCTUnwrap(operation[field: "query"]?[field: "allAnimals"] as? IR.EntityField)
+    let actual = subject.render(field: allAnimals)
+
+    // then
+    expect(actual).to(equalLineByLine(expected, atLine: 7, ignoringExtraLines: true))
+  }
+
+  func test__render_selection__givenInlineFragment_withSameTypeAndInclusionConditions_differentDeferConditions_shouldNotMergeSelectionsTogether() throws {
+    // given
+    schemaSDL = """
+    type Query {
+      allAnimals: [Animal!]
+    }
+
+    interface Animal {
+      genus: String!
+      species: String!
+    }
+
+    type Dog implements Animal {
+      genus: String!
+      species: String!
+    }
+    """
+
+    document = """
+    query TestOperation($filter: Boolean, $other: Boolean) {
+      allAnimals {
+        ... on Dog @include(if: $filter) @defer {
+          genus
+        }
+        ... on Dog @include(if: $filter) @defer(if: $other) {
+          species
+        }
+      }
+    }
+    """
+
+    let expected = """
+      public static var __selections: [ApolloAPI.Selection] { [
+        .field("__typename", String.self),
+        .include(if: "filter", [
+          .inlineFragment(AsDogIfFilter.self, deferred: true),
+          .inlineFragment(AsDogIfFilter.self, deferred: .if("other")),
```

Comment:
This looks like it's going to generate two separate selection sets both named `AsDogIfFilter`, which is not going to compile. We're going to need to handle this more carefully.